src/app/data-explorer/data-specification.service.ts

```diff
@@ -185,6 +189,40 @@ export class DataSpecificationService {
       );
   }
 
+  /**
+   * Opens a dialog for the user to set the specification
+   * as readable by any authenticated users or not.
+   *
+   * @param shared whether the {@link DataSpecification} is
+   * currently readable by any authenticated users or not.
+   * @returns an observable containing a {@link ShareDataSpecificationDialogResponse}
+   */
+  shareWithDialog(shared: boolean): Observable<ShareDataSpecificationDialogResponse> {
+    const user = this.security.getSignedInUser();
+    if (!user) return EMPTY;
+
+    const dialogData: ShareDataSpecificationDialogOptions = {
+      sharedWithCommunity: shared,
+    };
+
+    return this.dialogs
+      .shareWithCommunity(dialogData)
+      .afterClosed()
+      .pipe(
+        filter((response: any) => !!response),
+        map((response: ShareDataSpecificationDialogResponse) => {
```

Comment:
Is this `map()` RxJs operator needed since it just returns the same object back? Looks like it could be removed and still work.

Reply:
Fixed